pr_agent/algo/pr_processing.py

```diff
@@ -78,9 +78,9 @@
     return final_diff
 
 
-def pr_generate_extended_diff(pr_languages: list, token_handler: TokenHandler,
-                              add_line_numbers_to_hunks: bool) -> \
-        Tuple[list, int]:
+def pr_generate_extended_diff(pr_languages: list,
+                              token_handler: TokenHandler,
+                              add_line_numbers_to_hunks: bool) -> Tuple[list, int, list]:
```

Comment:
**Suggestion:** Consider adding a docstring to the 'pr_generate_extended_diff' function to explain its purpose, parameters, and return value.
```suggestion
def pr_generate_extended_diff(pr_languages: list,
                              token_handler: TokenHandler,
                              add_line_numbers_to_hunks: bool) -> Tuple[list, int, list]:
    """
    Generates an extended diff for the PR.

    Args:
        pr_languages (list): The main languages used in the PR.
        token_handler (TokenHandler): The handler for counting tokens.
        add_line_numbers_to_hunks (bool): Whether to add line numbers to the hunks in the diff.

    Returns:
        Tuple[list, int, list]: The extended patches, total tokens, and tokens per patch.
    """
```